lib/ProducerImpl.cc

```diff
@@ -172,6 +172,8 @@ void ProducerImpl::connectionFailed(Result result) {
 
 void ProducerImpl::handleCreateProducer(const ClientConnectionPtr& cnx, Result result,
                                         const ResponseData& responseData) {
+    Lock lock(mutex_);
```

Comment:
Could you explain more about the race condition you mentioned in this PR? I agree it does not affect performance significantly, but it should be reasonable to use a lock here.

Reply:
Thanks for your explanation. I missed the fact that you moved the `mutex` in line 187 here.